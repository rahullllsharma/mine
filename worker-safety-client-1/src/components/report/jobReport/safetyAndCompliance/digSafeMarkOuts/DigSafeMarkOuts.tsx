import type { DailyReportInputs } from "@/types/report/DailyReportInputs";
import type { SafetyAndComplianceProps } from "@/container/report/safetyAndCompliance/SafetyAndCompliance";
import { Controller, useFormContext } from "react-hook-form";

import ReportSectionHeader from "@/components/layout/report/reportSectionHeader/ReportSectionHeader";
import ReportSectionBlock from "@/components/layout/report/reportSectionBlock/ReportSectionBlock";
import ReportSectionWrapper from "@/components/layout/report/reportSectionWrapper/ReportSectionWrapper";
import FieldTextArea from "@/components/shared/field/fieldTextArea/FieldTextArea";
import FieldInput from "@/components/shared/field/fieldInput/FieldInput";
import FieldRadioGroup from "@/components/shared/field/fieldRadioGroup/FieldRadioGroup";
import { notApplicableRadioGroupOptions } from "@/radioGroupConstants";
import { FieldRules } from "@/components/shared/field/FieldRules";
import { getDefaultOption } from "../utils";

const rules = FieldRules.REQUIRED;

export type DigSafeMarkOutsInputs = {
  markOutsVerified: string;
  markOutsVerifiedNotes: string;
  facilitiesLocatedAndExposed: string;
  facilitiesLocatedAndExposedNotes: string;
  digSafeMarkOutsLocation: string;
};

const digSafeMarkOutsFormInputPrefix = "safetyAndCompliance.digSafeMarkOuts";

export default function DigSafeMarkOuts({
  isCompleted,
}: SafetyAndComplianceProps): JSX.Element {
  const {
    getValues,
    formState: { errors },
  } = useFormContext<Pick<DailyReportInputs, "safetyAndCompliance">>();

  return (
    <ReportSectionWrapper>
      <ReportSectionHeader title="Digsafe / Mark Outs" />

      <ReportSectionBlock isInner>
        <Controller
          name={`${digSafeMarkOutsFormInputPrefix}.markOutsVerified`}
          rules={rules}
          render={({ field: { onChange } }) => (
            <FieldRadioGroup
              label="Were all Mark Outs verified?"
              required
              options={notApplicableRadioGroupOptions}
              defaultOption={getDefaultOption(
                notApplicableRadioGroupOptions,
                getValues(`${digSafeMarkOutsFormInputPrefix}.markOutsVerified`)
              )}
              error={
                errors?.safetyAndCompliance?.digSafeMarkOuts?.markOutsVerified
                  ?.message
              }
              hasError={
                !!errors?.safetyAndCompliance?.digSafeMarkOuts?.markOutsVerified
              }
              onSelect={onChange}
              readOnly={isCompleted}
            />
          )}
        />
        <Controller
          name={`${digSafeMarkOutsFormInputPrefix}.markOutsVerifiedNotes`}
          render={({ field }) => (
            <FieldTextArea
              {...field}
              label="Notes"
              initialValue={getValues(
                `${digSafeMarkOutsFormInputPrefix}.markOutsVerifiedNotes`
              )}
              readOnly={isCompleted}
            />
          )}
        />
      </ReportSectionBlock>

      <ReportSectionBlock isInner>
        <Controller
          name={`${digSafeMarkOutsFormInputPrefix}.facilitiesLocatedAndExposed`}
          rules={rules}
          render={({ field: { onChange } }) => (
            <FieldRadioGroup
              label="Were all Facilities located and exposed prior to excavation?"
              required
              options={notApplicableRadioGroupOptions}
              defaultOption={getDefaultOption(
                notApplicableRadioGroupOptions,
                getValues(
                  `${digSafeMarkOutsFormInputPrefix}.facilitiesLocatedAndExposed`
                )
              )}
              error={
                errors?.safetyAndCompliance?.digSafeMarkOuts
                  ?.facilitiesLocatedAndExposed?.message
              }
              hasError={
                !!errors?.safetyAndCompliance?.digSafeMarkOuts
                  ?.facilitiesLocatedAndExposed
              }
              onSelect={onChange}
              readOnly={isCompleted}
            />
          )}
        />
        <Controller
          name={`${digSafeMarkOutsFormInputPrefix}.facilitiesLocatedAndExposedNotes`}
          render={({ field }) => (
            <FieldTextArea
              {...field}
              label="Notes"
              initialValue={getValues(
                `${digSafeMarkOutsFormInputPrefix}.facilitiesLocatedAndExposedNotes`
              )}
              readOnly={isCompleted}
            />
          )}
        />
      </ReportSectionBlock>

      <ReportSectionBlock>
        <Controller
          name={`${digSafeMarkOutsFormInputPrefix}.digSafeMarkOutsLocation`}
          rules={rules}
          render={({ field }) => (
            <FieldInput
              {...field}
              label="Digsafe location?"
              required
              error={
                errors?.safetyAndCompliance?.digSafeMarkOuts
                  ?.digSafeMarkOutsLocation?.message
              }
              value={getValues(
                `${digSafeMarkOutsFormInputPrefix}.digSafeMarkOutsLocation`
              )}
              readOnly={isCompleted}
            />
          )}
        />
      </ReportSectionBlock>
    </ReportSectionWrapper>
  );
}
