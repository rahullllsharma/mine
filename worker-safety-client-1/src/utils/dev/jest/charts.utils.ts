import type { RiskLevelByDate } from "@/components/charts/dailyRiskHeatmap/types";
import { act } from "@testing-library/react";
import { RiskLevel } from "@/components/riskBadge/RiskLevel";

export function chartRenderHelper(): () => void {
  // set up a listener so we can force the width/height
  // this is required to get the chart to render in its ResponsiveWrapper
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let listener: ((rect: any) => void) | undefined;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (window as any).ResizeObserver = class ResizeObserver {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    constructor(ls: any) {
      listener = ls;
    }
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    observe() {}
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    disconnect() {}
  };

  function triggerResize(): void {
    act(() => {
      listener &&
        listener([
          {
            contentRect: {
              x: 0,
              y: 0,
              width: 800,
              height: 800,
              top: 100,
              bottom: 0,
              left: 100,
              right: 0,
            },
          },
        ]);
    });
  }
  return triggerResize;
}

export const sampleHeatmapDates = {
  startDate: "2022-01-29",
  endDate: "2022-02-03",
};

// consumed by the DailyRiskHeatmap
export const sampleRiskLevelByDate: RiskLevelByDate[] = [
  {
    date: "2022-01-29",
    riskLevel: RiskLevel.HIGH,
  },
  {
    date: "2022-01-30",
    riskLevel: RiskLevel.MEDIUM,
  },
  {
    date: "2022-01-31",
    riskLevel: RiskLevel.UNKNOWN,
  },
  {
    date: "2022-02-01",
    riskLevel: RiskLevel.LOW,
  },
  {
    date: "2022-02-02",
    riskLevel: RiskLevel.RECALCULATING,
  },
  {
    date: "2022-02-03",
    riskLevel: RiskLevel.UNKNOWN,
  },
];
