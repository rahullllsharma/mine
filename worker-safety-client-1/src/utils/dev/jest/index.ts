export * from "./intl.utils";
export * from "./testing.utils";
export * from "./charts.utils";
