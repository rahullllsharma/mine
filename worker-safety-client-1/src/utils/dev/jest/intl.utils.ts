/** Mock the default locale when using the Intl.DateTimeFormat class */
export function mockIntlDateTimeFormatLocale(locale: string): void {
  const { DateTimeFormat } = Intl;
  jest
    .spyOn(global.Intl, "DateTimeFormat")
    .mockImplementation((_, options) => new DateTimeFormat(locale, options));
}
