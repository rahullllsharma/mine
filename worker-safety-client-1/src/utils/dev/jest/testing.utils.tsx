/* istanbul ignore file */
import type { ReactNode } from "react";
import type { PropsWithClassName } from "@/types/Generic";
import { fireEvent, render, screen } from "@testing-library/react";
import { FormProvider, useForm } from "react-hook-form";
import { useTenantStore } from "@/store/tenant/useTenantStore.store";
import { TenantMock } from "@/store/tenant/utils/tenantMock";

export const formTemplate = (
  component: ReactNode,
  defaultValues: { [key: string]: any } = {}
) => {
  function Wrapper({ children }: PropsWithClassName): JSX.Element {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  }
  return <Wrapper>{component}</Wrapper>;
};

export const formRender = (
  component: ReactNode,
  defaultValues: { [key: string]: any } = {}
) => {
  return render(formTemplate(component, defaultValues));
};

// to overcome react-select limitation for testing, of not responding to click events
export const openSelectMenu = (role = "combobox", name?: RegExp): void => {
  const inputElement = screen.getByRole(role, { name });
  fireEvent.keyDown(inputElement, { key: "ArrowDown", code: "ArrowDown" });
};

export const mockTenantStore = (mock = TenantMock) => {
  const { setTenant } = useTenantStore.getState();
  setTenant(mock);
};
