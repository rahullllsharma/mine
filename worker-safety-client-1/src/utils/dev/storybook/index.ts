export * from "./WrapperForm.utils";
export * from "./WrapperModal.utils";
export * from "./WrapperScroll.utils";
