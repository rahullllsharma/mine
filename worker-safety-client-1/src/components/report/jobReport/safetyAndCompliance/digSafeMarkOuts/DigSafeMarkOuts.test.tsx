import type { DailyReportInputs } from "@/types/report/DailyReportInputs";
import { formRender } from "@/utils/dev/jest";
import DigSafeMarkOuts from "./DigSafeMarkOuts";

const emptyStateValues: DailyReportInputs = {
  safetyAndCompliance: {
    digSafeMarkOuts: {
      markOutsVerified: "",
      markOutsVerifiedNotes: "",
      facilitiesLocatedAndExposed: "",
      facilitiesLocatedAndExposedNotes: "",
      digSafeMarkOutsLocation: "",
    },
  },
};

const prefilledValues: DailyReportInputs = {
  safetyAndCompliance: {
    digSafeMarkOuts: {
      markOutsVerified: "-1",
      markOutsVerifiedNotes: "N/A",
      facilitiesLocatedAndExposed: "1",
      facilitiesLocatedAndExposedNotes: "Yes",
      digSafeMarkOutsLocation: "No additional notes",
    },
  },
};

describe(DigSafeMarkOuts.name, () => {
  it("should display the DigsafeMarkOuts section without defaults", () => {
    const { asFragment } = formRender(<DigSafeMarkOuts />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the DigsafeMarkOuts section with empty state values", () => {
    const { asFragment } = formRender(<DigSafeMarkOuts />, emptyStateValues);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the DigsafeMarkOuts section with prefilled values", () => {
    const { asFragment } = formRender(<DigSafeMarkOuts />, prefilledValues);
    expect(asFragment()).toMatchSnapshot();
  });
});
