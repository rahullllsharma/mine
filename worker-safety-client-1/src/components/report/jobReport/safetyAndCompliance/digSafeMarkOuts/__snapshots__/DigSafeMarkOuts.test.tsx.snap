// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DigSafeMarkOuts should display the DigsafeMarkOuts section with empty state values 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <div
      class="mb-4 text-neutral-shade-100"
    >
      <h5
        class="font-semibold"
      >
        Digsafe / Mark Outs
      </h5>
    </div>
    <div
      class="flex flex-col mb-4 gap-4 bg-brand-gray-10 p-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Were all Mark Outs verified? *
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-15"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-17"
            class="flex items-center"
            id="headlessui-radiogroup-option-16"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-17"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-19"
            class="flex items-center"
            id="headlessui-radiogroup-option-18"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-19"
            >
              Yes
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-21"
            class="flex items-center"
            id="headlessui-radiogroup-option-20"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-21"
            >
              N/A
            </label>
          </div>
        </div>
      </div>
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Notes
        </label>
        <textarea
          aria-label="FieldTextArea"
          class="w-full max-h-36 text-base text-neutral-shade-100 outline-none focus:border-brand-gray-60 disabled:cursor-not-allowed min-h-12 border border-neutral-shade-37 rounded resize-y py-1 px-2"
          name="safetyAndCompliance.digSafeMarkOuts.markOutsVerifiedNotes"
        />
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4 bg-brand-gray-10 p-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Were all Facilities located and exposed prior to excavation? *
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-22"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-24"
            class="flex items-center"
            id="headlessui-radiogroup-option-23"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-24"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-26"
            class="flex items-center"
            id="headlessui-radiogroup-option-25"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-26"
            >
              Yes
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-28"
            class="flex items-center"
            id="headlessui-radiogroup-option-27"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-28"
            >
              N/A
            </label>
          </div>
        </div>
      </div>
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Notes
        </label>
        <textarea
          aria-label="FieldTextArea"
          class="w-full max-h-36 text-base text-neutral-shade-100 outline-none focus:border-brand-gray-60 disabled:cursor-not-allowed min-h-12 border border-neutral-shade-37 rounded resize-y py-1 px-2"
          name="safetyAndCompliance.digSafeMarkOuts.facilitiesLocatedAndExposedNotes"
        />
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Digsafe location? *
        </label>
        <div
          class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
        >
          <input
            aria-errormessage=""
            aria-invalid="false"
            class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
            name="safetyAndCompliance.digSafeMarkOuts.digSafeMarkOutsLocation"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DigSafeMarkOuts should display the DigsafeMarkOuts section with prefilled values 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <div
      class="mb-4 text-neutral-shade-100"
    >
      <h5
        class="font-semibold"
      >
        Digsafe / Mark Outs
      </h5>
    </div>
    <div
      class="flex flex-col mb-4 gap-4 bg-brand-gray-10 p-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Were all Mark Outs verified? *
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-29"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-31"
            class="flex items-center"
            id="headlessui-radiogroup-option-30"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-31"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-33"
            class="flex items-center"
            id="headlessui-radiogroup-option-32"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-33"
            >
              Yes
            </label>
          </div>
          <div
            aria-checked="true"
            aria-labelledby="headlessui-label-35"
            class="flex items-center"
            id="headlessui-radiogroup-option-34"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            >
              <div
                class="w-2.5 h-2.5 rounded-xl bg-brand-urbint-40 shadow-5"
              />
            </div>
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-35"
            >
              N/A
            </label>
          </div>
        </div>
      </div>
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Notes
        </label>
        <textarea
          aria-label="FieldTextArea"
          class="w-full max-h-36 text-base text-neutral-shade-100 outline-none focus:border-brand-gray-60 disabled:cursor-not-allowed min-h-12 border border-neutral-shade-37 rounded resize-y py-1 px-2"
          name="safetyAndCompliance.digSafeMarkOuts.markOutsVerifiedNotes"
          style="height: 0px;"
        >
          N/A
        </textarea>
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4 bg-brand-gray-10 p-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Were all Facilities located and exposed prior to excavation? *
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-36"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-38"
            class="flex items-center"
            id="headlessui-radiogroup-option-37"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-38"
            >
              No
            </label>
          </div>
          <div
            aria-checked="true"
            aria-labelledby="headlessui-label-40"
            class="flex items-center"
            id="headlessui-radiogroup-option-39"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            >
              <div
                class="w-2.5 h-2.5 rounded-xl bg-brand-urbint-40 shadow-5"
              />
            </div>
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-40"
            >
              Yes
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-42"
            class="flex items-center"
            id="headlessui-radiogroup-option-41"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-42"
            >
              N/A
            </label>
          </div>
        </div>
      </div>
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Notes
        </label>
        <textarea
          aria-label="FieldTextArea"
          class="w-full max-h-36 text-base text-neutral-shade-100 outline-none focus:border-brand-gray-60 disabled:cursor-not-allowed min-h-12 border border-neutral-shade-37 rounded resize-y py-1 px-2"
          name="safetyAndCompliance.digSafeMarkOuts.facilitiesLocatedAndExposedNotes"
          style="height: 0px;"
        >
          Yes
        </textarea>
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Digsafe location? *
        </label>
        <div
          class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
        >
          <input
            aria-errormessage=""
            aria-invalid="false"
            class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
            name="safetyAndCompliance.digSafeMarkOuts.digSafeMarkOutsLocation"
            type="text"
            value="No additional notes"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DigSafeMarkOuts should display the DigsafeMarkOuts section without defaults 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <div
      class="mb-4 text-neutral-shade-100"
    >
      <h5
        class="font-semibold"
      >
        Digsafe / Mark Outs
      </h5>
    </div>
    <div
      class="flex flex-col mb-4 gap-4 bg-brand-gray-10 p-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Were all Mark Outs verified? *
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-7"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-1"
            class="flex items-center"
            id="headlessui-radiogroup-option-2"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-1"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-3"
            class="flex items-center"
            id="headlessui-radiogroup-option-4"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-3"
            >
              Yes
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-5"
            class="flex items-center"
            id="headlessui-radiogroup-option-6"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-5"
            >
              N/A
            </label>
          </div>
        </div>
      </div>
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Notes
        </label>
        <textarea
          aria-label="FieldTextArea"
          class="w-full max-h-36 text-base text-neutral-shade-100 outline-none focus:border-brand-gray-60 disabled:cursor-not-allowed min-h-12 border border-neutral-shade-37 rounded resize-y py-1 px-2"
          name="safetyAndCompliance.digSafeMarkOuts.markOutsVerifiedNotes"
        />
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4 bg-brand-gray-10 p-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Were all Facilities located and exposed prior to excavation? *
        </label>
        <div
          class="flex flex-wrap flex-row gap-6"
          id="headlessui-radiogroup-14"
          role="radiogroup"
        >
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-8"
            class="flex items-center"
            id="headlessui-radiogroup-option-9"
            role="radio"
            tabindex="0"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-8"
            >
              No
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-10"
            class="flex items-center"
            id="headlessui-radiogroup-option-11"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-10"
            >
              Yes
            </label>
          </div>
          <div
            aria-checked="false"
            aria-labelledby="headlessui-label-12"
            class="flex items-center"
            id="headlessui-radiogroup-option-13"
            role="radio"
            tabindex="-1"
          >
            <div
              class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
            />
            <label
              class="text-neutral-shade-100 text-base ml-2"
              id="headlessui-label-12"
            >
              N/A
            </label>
          </div>
        </div>
      </div>
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Notes
        </label>
        <textarea
          aria-label="FieldTextArea"
          class="w-full max-h-36 text-base text-neutral-shade-100 outline-none focus:border-brand-gray-60 disabled:cursor-not-allowed min-h-12 border border-neutral-shade-37 rounded resize-y py-1 px-2"
          name="safetyAndCompliance.digSafeMarkOuts.facilitiesLocatedAndExposedNotes"
        />
      </div>
    </div>
    <div
      class="flex flex-col mb-4 gap-4"
    >
      <div>
        <label
          class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
        >
          Digsafe location? *
        </label>
        <div
          class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
        >
          <input
            aria-errormessage=""
            aria-invalid="false"
            class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
            name="safetyAndCompliance.digSafeMarkOuts.digSafeMarkOutsLocation"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
